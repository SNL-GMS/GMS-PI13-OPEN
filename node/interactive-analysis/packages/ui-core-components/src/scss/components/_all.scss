@import './widgets/all';
@import './core-table';
@import './dialog';
@import './d3';
@import './divider';
@import './resizer';
@import './table-ag-grid-overrides.scss';
@import './table-overrides';
@import './table-cell';
@import './scroll-box';
