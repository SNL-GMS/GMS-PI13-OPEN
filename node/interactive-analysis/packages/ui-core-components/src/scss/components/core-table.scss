/// Blueprint Table
.core-table__container {
  .bp3-table-header .bp3-table-column-name {
    height: 100%;
  }
  .bp3-table-header .bp3-table-header-content {
    position: absolute;
    margin-left: 24px;
    top: 0;
    display: inline-flex;
  }
  .core-table__column-header {
    display: flex;
    align-items: center;
    width: calc(100% - 25px);
    & .bp3-icon.bp3-intent-primary {
      color: $core-main;
    }
    .force-right {
      margin-left: auto;
      margin-right: 0.5em;
    }
  }

  /// Tooltips
  .bp3-table-cell .bp3-popover-wrapper,
  .bp3-table-cell .bp3-popover-target {
    width: 100%;
  }
}
.core-table__column-header--default {
  font-size: 16px;
  font-weight: 500;
  letter-spacing: 1px;
}
