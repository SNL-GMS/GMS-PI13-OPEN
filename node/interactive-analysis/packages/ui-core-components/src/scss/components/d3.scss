.tooltip-body {
  padding: 0.5em;
  font-size: 1em;
  border-radius: 3px;
  background-color: $core-prominent;
  color: $core-main-inverted;
  opacity: 0.9;
}

.no-pointer {
  pointer-events: none;
}

.chart__axis {
  path {
    stroke: $core-soft;
  }
  pointer-events: all;
}

.chart__axis-label {
  fill: $core-prominent;
}

.chart-bar:hover {
  cursor: pointer;
}

.x-tick {
  font-size: 1em;
  font-family: 'Helvetica Neue', Helvetica, Arial, sans-serif, Icons16 !important;
  pointer-events: all;
  &--no-data {
    fill: $core-soft;
  }
}

.x-tick-container {
  pointer-events: all;
}
.chart-title {
  font-size: 1.5em;
  fill: $core-main;
}
.chart-subtitle {
  font-size: 1.25em;
  fill: $core-prominent;
  font-weight: 100;
  text-rendering: geometricPrecision;
}
