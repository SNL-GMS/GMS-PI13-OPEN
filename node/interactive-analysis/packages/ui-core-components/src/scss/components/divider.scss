@import '~@blueprintjs/core/lib/scss/variables.scss';

$expanded-touch-target: -0.25em;

.divider-container {
  height: 100%;
  width: 100%;
  display: flex;
  flex-direction: column;
  overflow: auto;
}

.bottom-component {
  flex-grow: 100;
}

.horizontal-divider {
  position: relative;
  padding-top: 0.75em;
  padding-bottom: 0.75em;
  display: flex;
  width: 100%;
  height: auto;
  cursor: row-resize;
  flex: 0 0 auto;
  align-items: center;
  margin: 0 10px;
  &__spacer {
    position: relative;
    width: 50%;
    margin: 0;
    border-top: 2px dashed $core-soft;
  }
  &__target {
    position: absolute;
    top: $expanded-touch-target;
    bottom: $expanded-touch-target;
    width: 100%;
    z-index: 1;
  }
  &--hint {
    .resize-handle {
      fill: $core-prominent;
    }
    .horizontal-divider__spacer {
      border-color: $core-prominent;
    }
  }
  &--highlighted {
    background-clip: content-box;
    background-color: $core-table-selection;
    .horizontal-divider__spacer {
      border-color: $core-prominent;
    }
    .resize-handle {
      fill: $core-main;
    }
  }
}

.resize-handle {
  fill: $core-prominent;
  margin: 0 0.5em;
}
