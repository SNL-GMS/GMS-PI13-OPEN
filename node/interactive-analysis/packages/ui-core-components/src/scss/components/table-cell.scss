.table-cell {
  @include gpuAccelerated;
  position: relative;
  width: 100%;
  box-sizing: border-box;
  display: flex;
  align-items: center;
  padding: 0 10px;
  font-size: 16px;
  font-weight: 400;
  color: $core-main;

  /// Table Value styles
  .table-cell__value {
    position: relative;
    z-index: 1;
    display: flex;
    width: 100%;
    align-items: center;
    color: $core-main;
    & span {
      overflow: hidden;
      white-space: nowrap;
      text-overflow: ellipsis;
    }
    &--numeric {
      justify-content: flex-end;
      flex-grow: 10;
      width: auto;
    }
    &--center {
      justify-content: center;
      flex-grow: 10;
      width: auto;
    }
  }

  .table-cell-container {
    padding: 2px 0;
    background-color: $core-transparent;
  }

  /// Selection
  @mixin selectedBorder($where...) {
    position: relative;
    &::after {
      @include fillAllSpaceAbsolutely;
      content: '';
      @if length($where) > 0 {
        @each $place in $where {
          border-#{$place}: 4px solid lighten($core-input-highlight, 10%);
        }
      } @else {
        border: 4px solid lighten($core-input-highlight, 10%);
      }
      z-index: 1;
      opacity: 1;
      background-color: transparent;
    }
  }

  .is-selected & {
    @include selectedBorder('top', 'bottom', 'left', 'right');
  }
  .is-selected &,
  .ag-row-selected & {
    border: transparent !important;
  }
}
