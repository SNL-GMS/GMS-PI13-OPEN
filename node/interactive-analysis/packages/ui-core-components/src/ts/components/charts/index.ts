import { BarChart } from './bar-chart';
import { LineChart } from './line-chart';
import * as ChartTypes from './types';

export { ChartTypes, LineChart, BarChart };
export { PieChart, PieChartProps } from './pie-chart';
