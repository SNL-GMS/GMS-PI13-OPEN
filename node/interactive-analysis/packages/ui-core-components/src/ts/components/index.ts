export * from './table';
export * from './ui-widgets';
export * from './divider';
export * from './dialog';
export * from './resizer';
export * from './center-icon';
export * from './charts';
export * from './scroll-box';
export * from './collapse';
export * from './non-ideal-state';
