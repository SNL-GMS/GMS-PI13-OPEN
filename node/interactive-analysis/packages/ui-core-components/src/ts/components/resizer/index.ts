export { Resizer } from './resizer';
export { BottomResizeHandle } from './resize-handle';
export { ResizeContainer } from './resize-container';
export { ResizeContext } from './resize-context';
export * from './types';
