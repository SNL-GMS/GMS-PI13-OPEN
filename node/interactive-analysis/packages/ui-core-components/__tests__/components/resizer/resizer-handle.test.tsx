import React from 'react';
import { BottomResizeHandle, ResizeHandleProps } from '../../../src/ts/components/resizer';

// tslint:disable-next-line:no-implicit-dependencies no-var-requires no-require-imports
const Enzyme = require('enzyme');

// set up window alert and open so we don't see errors
(window as any).alert = jest.fn();
(window as any).open = jest.fn();

describe('BottomResizeHandle', () => {
  const props: ResizeHandleProps = {
    onResizeEnd: jest.fn(),
    handleMouseMove: jest.fn()
  };
  const mockBottomResizeHandle = Enzyme.shallow(<BottomResizeHandle {...props} />);
  it('is exported', () => {
    expect(BottomResizeHandle).toBeDefined();
  });
  it('will match the snapshot', () => {
    expect(mockBottomResizeHandle).toMatchSnapshot();
  });
});
