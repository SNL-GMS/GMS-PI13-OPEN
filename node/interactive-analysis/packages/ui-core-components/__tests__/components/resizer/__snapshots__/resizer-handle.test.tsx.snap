// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BottomResizeHandle will match the snapshot 1`] = `
<ResizeHandle
  handleMouseMove={[MockFunction]}
  onResizeEnd={[MockFunction]}
  orientation="BOTTOM"
/>
`;
